### DISCUSSION TITLE
How to pass in the feed xml files only the first part of the post?

### DISCUSSION THREAD

**Discussion by ilg-ul** [facebook/docusaurus#9509] [Q&A]
I noticed that in the feed xml files, the entire content of the post is passed as the `<content:encoded>` element.

My posts are large and I would like to pass only the first part, up to the `<!-- truncate -->` line.

Initially I thought that I can trim the `content` of the post via `feedOptions.createFeedItems`, but this was not effective, since the `content` is not used, instead the generated html is read from a file.

**Comment by ilg-ul** [comment]
Any suggestion?

**Comment by slorber** [comment]
> but this was not effective, since the `content` is not used, instead the generated html is read from a file.

I don't know what you mean here. Show me the code of your attempt and try to explain better.

---

I'm not sure it's a good idea to reparse the html content string to truncate it (you can if you really want to, with cheerio for example), but you can use any blog post metadata or custom front matter to populate the feed item content quite easily:

```js
              createFeedItems: async (params) => {
                const feedItems = await params.defaultCreateFeedItems(params);
                return feedItems.map((feedItem, index) => {
                  const blogPost = params.blogPosts[index]!;
                  return {
                    ...feedItem,
                    content: blogPost.metadata.description,
                  };
                });
              },
```

![CleanShot 2023-11-11 at 17 39 51@2x](https://github.com/facebook/docusaurus/assets/749374/af89bdc9-34ba-49ed-ae62-c4438205aec5)
